### PR TITLE
[feature] #2257: Revoke<Role> emits RoleRevoked event

### PR DESCRIPTION
**Pull request by omkar-mohanty**
### Description of the Change
This PR makes `Revoke` role emit `RoleRevoked` instead of `Permission` event. 
### Issue
`Revoke` role emits `Permission` events which is a misleading since roles are a collection of permission events.
### Benefits
More accurate information about the user.
### Possible Drawbacks
Not sure about drawbacks because I am new to the codebase.

### CODE REVIEW

**Comment by omkar-mohanty** [comment]
I stumbled upon `impl<W: WorldTrait> Execute<W> for Unregister<Role>` which emits a `PermissionRemoved` event should I change that to `RoleRevoked` event?

**Comment by omkar-mohanty** [comment]
@appetrosyan This PR is ready for review

**Review by appetrosyan — DISMISSED** [review]

**Comment by omkar-mohanty** [comment]
Hello maintainers I have addressed the issue in #2257 which was originally from #2007 by adding `RoleRevoked` event implementation. This PR still does not implement proper event processing for `RoleRevoked` as noted by @appetrosyan but I would like to have this PR merged as it is. Thanks for reviewing my code.

**Comment by omkar-mohanty** [comment]
For the event processing issue I have created #2280

**Review by s8sato — CHANGES_REQUESTED** [review]
For symmetry we also need something like `RoleGranted` for `PermissionAdded`

File: data_model/src/events/data/filters.rs [hunk 1]
```diff
@@ -458,6 +458,10 @@ mod account {
         ByPermissionAdded,
         /// Filter by PermissionRemoved event
         ByPermissionRemoved,
+        /// Filter by RoleRevoked event
+        ByRoleRevoked,
+        /// Filter by RoleGranted event
+        ByRoleGranted,
         /// Filter by MetadataInserted event
         ByMetadataInserted,
         /// Filter by MetadataRemoved event
```
Comment (outoftardis) [inline data_model/src/events/data/filters.rs:467]:
```suggestion
        ByPermissionAdded,
        /// Filter by `PermissionRemoved` event
        ByPermissionRemoved,
        /// Filter by `RoleRevoked` event
        ByRoleRevoked,
        /// Filter by `RoleGranted` event
        ByRoleGranted,
        /// Filter by `MetadataInserted` event
        ByMetadataInserted,
        /// Filter by `MetadataRemoved` event
```

plus the same change for the lines above these

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**